### PR TITLE
Parens: Keep parens for non-identical infix operator pairs with same precedence

### PR DESCRIPTION
**Pull request by brianrourkeboll**
Another followup to #16079.

## Description

Keep parentheses for non-identical infix operator pairs that have the same precedence, e.g.:

```fsharp
f <| (g << h)
```

```fsharp
x + (y ++ z)
```

etc.

We still offer to remove them when both operators are the same, e.g.:

```fsharp
x + (y + z) // → x + y + z
```

_except_ for the left-associative "relational" operators (`=…`, `|…`, `&…`, `$…`, `>…`, `<…`, `!=…`, including `|>`, `<|`, `>>`, `<<`, etc.), where the order of operations is often arguably non-obvious:

```fsharp
f <| (g <| x)
```

```fsharp
x <> (y <> z)
```

```fsharp
x > (y < z)
```

## Checklist

- [x] Test cases added.
- [x]  Release notes entry updated.

### CODE REVIEW

File: src/Compiler/Service/ServiceAnalysis.fs
[the remark this answers is not in the document]
Comment (psfinaki) [reply src/Compiler/Service/ServiceAnalysis.fs:1]: Sure :)

**Review by psfinaki — APPROVED** [review]

**Comment by psfinaki** [comment]
@brianrourkeboll this can also go once the release notes conflict is resolved :)

@dotnet/fsharp-team-msft please take a look

**Comment by brianrourkeboll** [comment]
All right @psfinaki, I think this should finally be good now that the release notes PR has been merged in.

**Review by T-Gro — APPROVED** [review]

**Pull request merged**